extensions/amp-ad-network-dianomi-impl/0.1/amp-ad-network-dianomi-impl.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpA4A} from '../../amp-a4a/0.1/amp-a4a';
+
+export class AmpAdNetworkDianomiImpl extends AmpA4A {
+  /** @override */
+  getAdUrl() {
+    const id = this.element.getAttribute('data-request-param-id');
+    return `https://www.dianomi.com/smartads.pl?format=a4a&id=${id}`;
```

Comment:
If `data-request-param-id` is not specified, this will return the URL without an id which may lead an invalid ad request. Is that scenario handled somewhere else or should it be handled here?

Reply:
Good point, maybe just throw here since it is a required attr? If this is what you want you can use our `userAssert(id, 'some message')`

Reply:
userAssert will now be triggered if the request param ID is invalid or blank.